### PR TITLE
tidy up the template questions

### LINKED ISSUE #9: Tidy and Simplify
Along with https://github.com/epics-containers/services-template-helm/issues/9 I would also like to clean up this template.

There are too many questions and I'm not clear on how to answer them all so I feel this is too complicated for release to controls team.

I think an early question about is this for DLS could lead down a path of more defaults. Maybe.

### PR DESCRIPTION
**Pull request by gilesknap**
simplify the questions and add more help.

Points to note.
Target namespace, Apps namespace and Argo CD project are all the same string at DLS so I condensed down into a single question. If any external users want to set up ArgoCD we can recommend they use the same approach or generalize this template as necessary.

tested against i04 and p47.

Fixes #9 

example interaction:

<pre>This template will create a new repository which describes the set of IOCs and services that Argo CD will track on a Kubernetes cluster.

<font color="#5F87AF">🎤</font><b> A name for the group of IOC and service instances that this</b>
<b>   repository will track.</b>

<b>   At DLS this should be the short beamline name or the technical area</b>
<b>   for accelerator repos.</b>
<b>   </b><font color="#FFAF00"><b>i04</b></font>
<font color="#5F87AF">🎤</font><b> A One line description of the module</b>
<b>   </b><font color="#FFAF00"><b>i04 IOC Instances and Services Deployment Repository for Argo CD</b></font>
<font color="#5F87AF">🎤</font><b> Argocd server DNS Name.</b>

<b>   At DLS this is always &quot;argocd.diamond.ac.uk&quot;</b>
<b>   </b><font color="#FFAF00"><b>argocd.diamond.ac.uk</b></font>
<font color="#5F87AF">🎤</font><b> Cluster in which Argo CD Apps are deployed.</b>

<b>   At DLS this is always &quot;argus&quot;</b>
<b>   </b><font color="#FFAF00"><b>argus</b></font>
<font color="#5F87AF">🎤</font><b> The Kubernetes cluster where the IOCs and services will run.</b>

<b>   At DLS this should be &quot;{beamline shortname}&quot;, &quot;acastus&quot; for the</b>
<b>   accelerator or &quot;pollux&quot; for test beamlines.</b>
<b>   </b><font color="#FFAF00"><b>i04</b></font>
<font color="#5F87AF">🎤</font><b> Kubernetes namespace in which the IOCs and services will run.</b>

<b>   At DLS this should be &quot;{beamline shortname}-beamline&quot; or &quot;accelerator&quot;.</b>
<b>   </b><font color="#FFAF00"><b>i04-beamline</b></font>
<font color="#5F87AF">🎤</font><b> Git platform hosting this repository.</b>
<b>   </b><font color="#FFAF00"><b>gitlab.diamond.ac.uk</b></font>
<font color="#5F87AF">🎤</font><b> The DLS technical area</b>
<b>   </b><font color="#FFAF00"><b>beamline</b></font>
<font color="#5F87AF">🎤</font><b> Remote URI of this repository.</b>
<b>   </b><font color="#FFAF00"><b>https://gitlab.diamond.ac.uk/controls/containers/beamline/i04-deployment.git</b></font>
<font color="#5F87AF">🎤</font><b> Remote URI of the services repository that defines the IOCs and services,</b>
<b>   that this repository will track.</b>
<b>   </b><font color="#FFAF00"><b>https://gitlab.diamond.ac.uk/controls/containers/beamline/i04-services.git</b></font>
<font color="#5F87AF">🎤</font><b> The initial release of the services repository to track e.g. &quot;2024.12.1&quot; or &quot;main&quot;</b>
<b>   </b><font color="#FFAF00"><b>main</b></font>
<font color="#5F87AF">🎤</font><b> URL for centralized logging.</b>
<b>   </b><font color="#FFAF00"><b>DLS</b></font>

</pre>

### CODE REVIEW

File: copier.yml [hunk 1]
```diff
@@ -1,16 +1,20 @@
...
 
 ioc_group:
   type: str
-  help: Short name for the collection of services.
-  placeholder: e.g. "t01", "VA", "PS"
+  help: |
+    A name for the group of IOC and service instances that this
+       repository will track.
+
+       At DLS this should be the short beamline name or the technical area
```
Comment (marcelldls) [inline copier.yml:14]: When I made this template I avoided any "At DLS..." because I feel that discourages collaboration (suggests other facilities should be forking). My personal opinion is I would discourage becoming "Diamond Epics Containers".

There's definitely a trade off to be made because it seems to be confusing to onboard people more familiar with Diamond then they are of Containerisation. Is it an idea to move this repo rather to the DLS organisation and then make it very specific to our application?

File: copier.yml [hunk 2]
```diff
@@ -147,11 +136,11 @@ services_repo:
 
 services_release:
   type: str
-  help: The initial release of the services repository to track e.g. "2024.1.1"
+  help: The initial release of the services repository to track e.g. "2024.12.1" or "main"
```
Comment (marcelldls) [inline copier.yml:139]: main is not a release - maybe rephrase "release" -> git ref" to make it clear what is expected?

Comment (marcelldls) [inline copier.yml:139]: Im second guessing the use of "initial". Should updates to the release not be done by a copier update as best practice?

**Comment by GDYendell** [comment]
> 🎤 URL for centralized logging.
   DLS

What is this?

I think this is better. The only thing I would question is that it is halfway between being generic and being DLS-specific. I don't think this would be used by an external site as is, so we could either make it more convenient to use at DLS (more strict defaults instead of saying "at DLS ..."), or make it completely generic and then probably write some internal documentation to know what to use at DLS.

File: template/apps/values.yaml.jinja [hunk 3]
```diff
@@ -35,6 +35,8 @@ ec_services:
...
   #     enabled: set to false to stop a service
-  {{ ioc_group }}-ea-test-01:
-    enabled: true
-    targetRevision: {{ services_release }}
\ No newline at end of file
+
+  # example entry
+  #   {{ ioc_group }}-ea-test-01:
+  #     enabled: true
+  #     targetRevision: {{ services_release }}
```
Comment (marcelldls) [inline template/apps/values.yaml.jinja:42]: My personal feeling is that after I create a repo, I want to deploy it and know it works as is. Therefore I would prefer to not comment this out

File: template/apps.yaml.jinja [hunk 4]
```diff
@@ -5,10 +5,10 @@ metadata:
   finalizers:
     - resources-finalizer.argocd.argoproj.io
 spec:
-  project: {{ argocd_project }}
+  project: {{ cluster_namespace }}
```
Comment (marcelldls) [inline template/apps.yaml.jinja:8]: For general use - Im not sure its a safe assumption that the name of the project=namespace

File: copier.yml (hunk 2 above)
Comment (gilesknap) [reply copier.yml:139]: I dont think so - this is just what you will start with. I think that is what Initial means ?

File: template/apps/values.yaml.jinja (hunk 3 above)
Comment (gilesknap) [reply template/apps/values.yaml.jinja:42]: I have removed test from the default services repo - as per a previous conversation so it won't work if I uncomment that.

Happy to re-consider this.

File: template/apps.yaml.jinja (hunk 4 above)
Comment (gilesknap) [reply template/apps.yaml.jinja:8]: But it is at DLS and when we come to write docs for ArgoCD we should recommend this way of deploying it.

However I also take Gary's point about half way between specific and general. I did consider making this DLS only for the moment.

**Comment by gilesknap** [comment]
> I don't think this would be used by an external site as is, so we could either make it more convenient to use at DLS (more strict defaults instead of saying "at DLS ..."), or make it completely generic and then probably write some internal documentation to know what to use at DLS.

I think this is a valid point and I considered switching to totally DLS specific for this for the moment. We could always make a new more generic version if anyone ever ends up adopting.

Making it DLS specific removes around half of the questions

File: copier.yml (hunk 2 above)
Comment (marcelldls) [reply copier.yml:139]: Yes that is currently what is implied - but the question is should it? Should we not encourage updating the release rather through a copier update?

Comment (gilesknap) [reply copier.yml:139]: I feel that dirties the idea of what the template is for. But that's because I've not considered using this way before. But I think I prefer the idea that ec updates what we track, the template updates what the generic parts of the repo look like.

**Comment by gilesknap** [comment]
> > 🎤 URL for centralized logging.
> > DLS
> 
> What is this?

IT is the Graylog URL used when you type `ec log-history xxxx`

File: copier.yml (hunk 2 above)
Comment (gilesknap) [reply copier.yml]: I changed it to 'tag or branch'

File: template/apps/values.yaml.jinja (hunk 3 above)
Comment (marcelldls) [reply template/apps/values.yaml.jinja:42]: Ah okay I did not realize that this was chosen. Thats fine then

File: template/apps.yaml.jinja (hunk 4 above)
Comment (gilesknap) [reply template/apps.yaml.jinja:8]: resolving this and continuing discussion in main thread

File: copier.yml (hunk 1 above)
Comment (gilesknap) [reply copier.yml:14]: lets discuss the fully DLS vs fully generic vs somewhere in between issue in the main thread.

File: copier.yml (hunk 2 above)
Comment (marcelldls) [reply copier.yml]: Thats fine. Its not used in more places so copier probably wont break. I imagine ec only updating on a service level though so its two different places

**Comment by marcelldls** [comment]
> So I vote for sticking with what this PR has:
> 
> * mostly generic choices (but with namespaces and projects fixed as equal)
> * hints that describe what we do at DLS

I agree that the PR does this. Its a decent middle ground. Changes LGTM

**Comment by gilesknap** [comment]
Thanks @marcelldls. Merging.
